src/main/java/seedu/address/logic/commands/StackUndoRedo.java

```diff
@@ -10,48 +10,61 @@
 
 public class StackUndoRedo {
     private Stack<RedoableCommand> undoStack;
+    private Stack<String> commandUndoStack;
+
     private Stack<RedoableCommand> redoStack;
+    private Stack<String> commandRedoStack;
 
     /**
      * StackUndoRedo Class to enable storing of commands FIFO
      */
     public StackUndoRedo() {
         undoStack = new Stack<>();
         redoStack = new Stack<>();
+
+        commandRedoStack = new Stack<>();
+        commandUndoStack = new Stack<>();
     }
 
     /**
      * Pushes {@code command} onto the undo-stack if it is of type {@code UndoableCommand}. Clears the redo-stack
      * if {@code command} is not of type {@code UndoCommand} or {@code RedoCommand}.
      */
-    public void push(Command command) {
+    public void push(Command command, String commandName) {
         if (!(command instanceof UndoCommand) && !(command instanceof RedoCommand)) {
             redoStack.clear();
+            commandRedoStack.clear();
         }
 
         if (!(command instanceof RedoableCommand)) {
             return;
         }
-
+        commandUndoStack.add(commandName);
```

Comment:
I'm assuming this is to display:
![image](https://user-images.githubusercontent.com/70756909/161460001-b9e5e39f-9474-46ce-8847-13304b69788a.png), but I'm not sure how useful this is, since you can undo/redo the command instead of needing to know what command you just undid/redid anyways.

Would it be more useful to display the number of commands left to undo/redo? So just show a `Undo success! Commands left to undo: 2` or something like that

Reply:
I think I prefer displaying command though, because sometimes you would forget what command you were undoing/redoing, since it might be hard to pick out changes visually and you might be undoing a lot of commands in quick succession.

Reply:
Actually I don't think can change the error message. [(Link)](https://github.com/nus-cs2103-AY2122S2/forum/issues/238)

Reply:
> Actually I don't think can change the error message. [(Link)](https://github.com/nus-cs2103-AY2122S2/forum/issues/238)

Hm so in that case then @junlee1991 cannot change the error message to what it is right now, and should only keep it as `Undo success!` and `Redo success!`

Reply:
I needed to error message to validate whether is undoing the right command for debugging.

Reply:
> I needed to error message to validate whether is undoing the right command for debugging.

Yes I understand, but we can't change error messages as per v1.4. requirements so you'll have to revert back to what it was before.

Reply:
> I needed the error message to validate whether is undoing the right command for debugging.

I previously put a toString method for UndoRedoStack to print state of UndoStack and RedoStack.